nixos/modules/services/web-apps/dawarich.nix

```diff
@@ -0,0 +1,729 @@
+        };
+
+        systemd.services.dawarich-web = {
+          after = [
+            "network.target"
+          ]
+          ++ lib.optional needsGenCredentialsUnit "dawarich-init-credentials.service"
+          ++ commonUnits;
+          requires = lib.optional needsGenCredentialsUnit "dawarich-init-credentials.service" ++ commonUnits;
+          wantedBy = [ "dawarich.target" ];
+          description = "Dawarich web";
+          environment = env;
+          script = ''
+            ${loadCredentialsIntoEnv}
+
+            ${lib.getExe' cfg.package "rails"} server
+          '';
+          serviceConfig = {
+            Restart = "always";
+            RestartSec = 20;
+            LoadCredential = loadCredentials;
+            EnvironmentFile = cfg.extraEnvFiles;
+            WorkingDirectory = cfg.package;
+            # Runtime directory and mode
+            RuntimeDirectory = "dawarich-web";
+            RuntimeDirectoryMode = "0750";
+          }
+          // cfgService;
+        };
+
+        systemd.tmpfiles.settings."dawarich"."${dataDir}/imports/watched".d = {
+          group = "dawarich";
+          mode = "700";
+          user = "dawarich";
+        };
+
+        services.nginx = lib.mkIf cfg.configureNginx {
+          enable = true;
+          virtualHosts."${cfg.localDomain}" = {
+            root = "${cfg.package}/public/";
+
+            locations."/" = {
+              tryFiles = "$uri @proxy";
+            };
+
+            locations."@proxy" = {
+              proxyPass = "http://127.0.0.1:${toString cfg.webPort}";
+              recommendedProxySettings = true;
+              proxyWebsockets = true;
+            };
+          };
+        };
+
+        services.redis.servers.dawarich = lib.mkIf redisActuallyCreateLocally (
+          lib.mkMerge [
+            {
+              enable = true;
+            }
+            (lib.mkIf (!cfg.redis.enableUnixSocket) {
+              port = cfg.redis.port;
+            })
+          ]
+        );
+
+        services.postgresql = lib.mkIf databaseActuallyCreateLocally {
+          enable = true;
+          extensions = ps: with ps; [ postgis ];
+          ensureUsers = [
+            {
+              inherit (cfg.database) name;
+              ensureDBOwnership = true;
+            }
+          ];
+          ensureDatabases = [ cfg.database.name ];
+        };
+
+        systemd.services.postgresql-setup.serviceConfig.ExecStartPost =
+          let
+            # https://github.com/Freika/dawarich/blob/0.30.6/db/schema.rb#L15-L16
+            extensions = [
+              "plpgsql"
+              "postgis"
+            ];
+            sqlFile = pkgs.writeText "dawarich-postgres-setup.sql" ''
+              ${lib.concatMapStringsSep "\n" (ext: "CREATE EXTENSION IF NOT EXISTS \"${ext}\";") extensions}
+              ${lib.concatMapStringsSep "\n" (ext: "ALTER EXTENSION \"${ext}\" UPDATE;") extensions}
```

Comment:
This is not quite right, postgis is updated with `SELECT postgis_extensions_upgrade()`